test-network-function/lifecycle/suite.go

```diff
@@ -479,3 +485,36 @@ func testOwner(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func testImagePolicy(env *config.TestEnvironment) {
```

Comment:
Three general comments in here : 

1. you're assuming the loop will give you containers belonging to the same pod successively. in golang map, the order is random. so this logic will not work.
2. it's better to get the comment per container directly if possible.
3. the way you're looping here, you'll stop the test in the first failed container. It's better to find all the failed containers. -look at tainted kernel test as reference-

Reply:
Nice catch. @aabughosh Just put the original version back, where you loop through podUnderTest.

Reply:
ohh thanks salah didnt know about that i updated the loop to go into the pods